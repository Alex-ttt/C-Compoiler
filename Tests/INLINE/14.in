inline int fo(int x){
	int y = x;
	if (y > 0)
		return y;
	else
		return -y;
}

void main(){
	printf("%d %d %d", fo(-5), fo(6), fo(0));
}