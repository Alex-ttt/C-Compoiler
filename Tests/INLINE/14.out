 Symbols Table:
   fo  int function fo(
          x int
  )
    {
      Statements:
        y
      =
        x
--------------------------
      if

          y
        >
          0

      then

        {
          Statements:
          return
            y
--------------------------
        }
      else

        {
          Statements:
          return
            -
              y
--------------------------
        }
--------------------------
      Symbols Table:
              y int
    }
   main  void function main(
  )
    {
      Statements:
      {
        Statements:
          param0
        =
          -
            5
--------------------------
          y
        =
          param0
--------------------------
        if

            y
          >
            0

        then

          {
            Statements:
              inline_temp0
            =
              y
            goto: BLOCK_LABEL6
--------------------------
          BLOCK_LABEL5:
          }
        else

          {
            Statements:
              inline_temp0
            =
              -
                y
            goto: BLOCK_LABEL6
--------------------------
          BLOCK_LABEL4:
          }
--------------------------
      BLOCK_LABEL6:
        Symbols Table:
                  y int
      }
--------------------------
      {
        Statements:
          param0
        =
          6
--------------------------
          y
        =
          param0
--------------------------
        if

            y
          >
            0

        then

          {
            Statements:
              inline_temp1
            =
              y
            goto: BLOCK_LABEL9
--------------------------
          BLOCK_LABEL8:
          }
        else

          {
            Statements:
              inline_temp1
            =
              -
                y
            goto: BLOCK_LABEL9
--------------------------
          BLOCK_LABEL7:
          }
--------------------------
      BLOCK_LABEL9:
        Symbols Table:
                  y int
      }
--------------------------
      {
        Statements:
          param0
        =
          0
--------------------------
          y
        =
          param0
--------------------------
        if

            y
          >
            0

        then

          {
            Statements:
              inline_temp2
            =
              y
            goto: BLOCK_LABEL12
--------------------------
          BLOCK_LABEL11:
          }
        else

          {
            Statements:
              inline_temp2
            =
              -
                y
            goto: BLOCK_LABEL12
--------------------------
          BLOCK_LABEL10:
          }
--------------------------
      BLOCK_LABEL12:
        Symbols Table:
                  y int
      }
--------------------------
      printf(
        "%d %d %d"
      ,
          inline_temp0
        ,
          inline_temp1
        ,
          inline_temp2
      )
--------------------------
      Symbols Table:
              inline_temp0 int
              inline_temp1 int
              inline_temp2 int
    }
